app/values/hls_manifest.rb

```diff
@@ -7,11 +7,17 @@ class HlsManifest
   # returns the manifest referenced by file_metadata, mutated if needed.
   # @return [#to_s] Object whose #to_s method returns an HLS Manifest.
   def self.for(file_set:, file_metadata:, as: nil, auth_token: nil)
+    # Captions need a manifest to point to the VTT
+    return Caption.new(file_set: file_set, file_metadata: file_metadata, auth_token: auth_token) if file_metadata.caption?
     return unless file_metadata.hls_manifest?
+    # IIIF Manifests link to a "primary" HLS manifest, which contains links to the caption
+    # manifest and the video manifest. Build it dynamically because it's easier.
     if as == "stream"
       Primary.new(file_set: file_set, file_metadata: file_metadata, auth_token: auth_token)
     else
-      new(file_set: file_set, file_metadata: file_metadata, auth_token: auth_token)
+      # If not asked for the primary manifest, just render the video HLS manifest
+      # ffmpeg created (with auth_tokens if appropriate)
+      new(file_set: file_set, file_metadata: file_metadata, auth_token: auth_token) unless as == "stream"
```

Comment:
redundant `unless` here

Reply:
Oof, thank you. Postfix conditionals are sneaky. 🥷